flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/ExecutionDeploymentReconcilerImplTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmaster;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
+import org.apache.flink.runtime.executiongraph.ExecutionAttemptID;
+import org.apache.flink.runtime.taskexecutor.ExecutionDeploymentReport;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.concurrent.CompletableFuture;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for {@link ExecutionDeploymentReconcilerImpl}.
+ */
+public class ExecutionDeploymentReconcilerImplTest extends TestLogger {
+
+	@Test
+	public void testMatchingDeployments() throws Exception {
+		runTest((reconciler, missingFuture, unknownFuture) -> {
+			ResourceID resourceId = ResourceID.generate();
+			ExecutionAttemptID attemptId = new ExecutionAttemptID();
+
+			reconciler.reconcileExecutionDeployments(
+				resourceId,
+				new ExecutionDeploymentReport(Collections.singleton(attemptId)),
+				Collections.singleton(attemptId));
+
+			assertFalse(missingFuture.isDone());
+			assertFalse(unknownFuture.isDone());
+		});
+	}
```

Comment:
It's mainly a personal preference but I think tests are easier to understand if one know what's going on. With `runTest` a lot of details are being hidden. For example, the below suggestion makes it a bit easier to understand what `missingFuture` and `unknownFuture` actually are.

```suggestion
       @Test
	public void test() {
		final TestingExecutionDeploymentReconciliationHandler handler = new TestingExecutionDeploymentReconciliationHandler();
		final ExecutionDeploymentReconciler reconciler = new ExecutionDeploymentReconcilerImpl(handler);
		ResourceID resourceId = ResourceID.generate();
		ExecutionAttemptID attemptId = new ExecutionAttemptID();

		reconciler.reconcileExecutionDeployments(
			resourceId,
			new ExecutionDeploymentReport(Collections.singleton(attemptId)),
			Collections.singleton(attemptId));

		assertThat(handler.getMissingDeployments(), is(empty()));
		assertFalse(handler.getUnknownDeployments(), is(empty()));
	}
```